File: tests/java/Fresh.java


class Fresh {

    /*@ allocates \result;
      @ ensures \fresh(\result);
      @*/
    static Fresh create() {
        return new Fresh ();
    }

    void test () {
        Fresh f = create ();
        //@ assert this != f;
    }
}




/*
Local Variables:
compile-command: "make Fresh.why3ide"
End:
*/

File: src/frama-c-plugin/tests/jessie/fresh.c




/*@ assigns \nothing;
  @ allocates \result;
  @ ensures \valid(\result) && \fresh(\result,sizeof(int));
  @*/
int* f(void);


void g() {
   int* p;
   p = f();
   *p = 0;
}

/* 
Local Variables:
compile-command: "LC_ALL=C make fresh.why3ide"
End:
*/
